src/perftest_communication.c

```diff
@@ -1124,35 +1132,36 @@ int rdma_server_connect(struct pingpong_context *ctx,
 	struct rdma_conn_param conn_param;
 	struct addrinfo hints;
 	char *service;
-	struct sockaddr_in sin;
+	struct sockaddr *sin;
```

Comment:
keep a copy of the destination address, so `res` can be freed earlier, `freeaddrinfo(res)` needn't to be called more than once.
suggest changing it to `struct sockaddr_storage sin`, and casting it `struct sockaddr *` when used

Reply:
Sorry, I want to be sure that I understand you correctly, do you mean to keep the old implementation ? (memcpy(&sin, res->ai_addr, sizeof(sin))) and then free the res regularly right after in addition to changing sin to sockaddr_storage and casting when needed?

Reply:
yes, keep the old implementation.
refer to my another comment:
https://github.com/linux-rdma/perftest/pull/218/files#r1254053570